apiclient/harvester_api/models/addons.py

```diff
@@ -0,0 +1,133 @@
+from json import loads, dumps
+
+from yaml import safe_load, safe_dump
+
+
+class _BaseConfig:
+    def __init__(self, data):
+        self._val = data
+        self.limits = data['resources']['limits']
+        self.requests = data['resources']['requests']
+
+    def sync(self):
+        """ To sync attributes into dictionary"""
+
+
+class BaseAddonSpec:
+    """ Base class for instance check and create """
+    _name = ""  # to point out the name of addon
+    _id = ""
+    _enable = False  # the addon is enable or not
+
+    def __init__(self, value=None, enabled=False):
+        self.value = value or dict()
+        self.enable = enabled
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}({self.value})"
+
+    @property
+    def enable(self):
+        return self._enable
+
+    @enable.setter
+    def enable(self, val):
+        self._enable = val
+
+    def to_dict(self, data):
```

Comment:
`data` argument is not been used?

Reply:
ref to https://github.com/harvester/tests/blob/7d41a56d02f276091e045696caf31511dd03dfbf/apiclient/harvester_api/models/settings.py#L129-L131, the argument is the placeholder for sub-classes in case they need the value from outer.